packages/peregrine/src/hooks/useDOMQuery.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const findDOMElements = queryString =>
+    document.querySelectorAll(queryString) || [];
+
+/**
+ * A custom hook that will return the array of elements that
+ * match the queryString. document.querySelectorAll is used
+ * to find all elements.
+ *
+ * @param {string} queryString
+ * @return Array<HTMLElement> and {
+ *  setInnerHTML: (string) => void,
+ *  setInnerText: (string) => void,
+ *  setAttribute: (string, any) => void
+ * }
+ */
+export const useDOMQuery = queryString => {
+    const [elements, setElements] = useState([]);
+    useEffect(() => {
+        setElements(findDOMElements(queryString));
+    }, []);
+    const setInnerHTML = useCallback(
+        content => {
+            elements.forEach(element => {
+                element.innerHTML = content || element.innerHTML;
+            });
+        },
+        [elements]
+    );
+    const setInnerText = useCallback(
```

Comment:
You may understand hooks better than me. The collection of elements will likely change identity every time. Why do we need `useCallback` here?

Reply:
Collection of elements will only change on the component creation and component mount but component itself might be re-rendered multiple times depending on the parent's implementation. I realized we don't need to recreate all the helper functions on every render. This is based on the assumption that state variables references are memoized internally when we use `useState` and only change when its setter is used. Correct me if I am wrong.

Reply:
When I said component in the above context, I meant the hook.